File: examples/python/load_json.py
import json
import os

# Path to the data relative to the project root
RANKS_PATH = '../../data/json/thai_military_ranks.json'
GROUPED_RANKS_PATH = '../../data/json/thai_military_ranks_grouped.json'

def load_example():
    # Adjust path if running from examples/python
    base_dir = os.path.dirname(__file__)
    ranks_file = os.path.join(base_dir, RANKS_PATH)
    grouped_file = os.path.join(base_dir, GROUPED_RANKS_PATH)

    print("--- Loading Flat JSON ---")
    with open(ranks_file, 'r', encoding='utf-8') as f:
        ranks = json.load(f)
        print(f"Loaded {len(ranks)} ranks.")
        print(f"First rank: {ranks[0]['rank_th']} ({ranks[0]['rank_en']})")

    print("\n--- Loading Grouped JSON ---")
    with open(grouped_file, 'r', encoding='utf-8') as f:
        grouped = json.load(f)
        for branch, items in grouped.items():
            print(f"{branch}: {len(items)} items")

if __name__ == "__main__":
    load_example()

File: examples/javascript/load_json.js
const fs = require('fs');
const path = require('path');

const RANKS_PATH = path.join(__dirname, '../../data/json/thai_military_ranks.json');

try {
  const data = fs.readFileSync(RANKS_PATH, 'utf8');
  const ranks = JSON.parse(data);

  console.log('--- JavaScript JSON Load Example ---');
  console.log(`Loaded ${ranks.length} ranks.`);
  
  const gen = ranks.find(r => r.id === 'RTA-O-01');
  if (gen) {
    console.log(`Found Rank: ${gen.rank_th} (${gen.rank_en})`);
  }
} catch (err) {
  console.error('Error loading JSON:', err);
}
